packages/template-retail-react-app/app/components/shopper-agent/index.jsx

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useEffect} from 'react'
+import useScript from '@salesforce/retail-react-app/app/hooks/use-script'
+import {useUsid} from '@salesforce/commerce-sdk-react'
+import PropTypes from 'prop-types'
+import {useTheme} from '@salesforce/retail-react-app/app/components/shared/ui'
+
+const onClient = typeof window !== 'undefined'
+
+// Function to initialize embedded messaging
+const initEmbeddedMessaging = (
+    salesforceOrgId,
+    embeddedServiceDeploymentName,
+    embeddedServiceDeploymentUrl,
+    scrt2Url,
+    locale
+) => {
+    try {
+        if (
+            onClient &&
+            window.embeddedservice_bootstrap &&
+            window.embeddedservice_bootstrap.settings
+        ) {
+            window.embeddedservice_bootstrap.settings.language = locale
+            window.embeddedservice_bootstrap.settings.disableStreamingResponses = true
+            window.embeddedservice_bootstrap.init(
+                salesforceOrgId,
+                embeddedServiceDeploymentName,
+                embeddedServiceDeploymentUrl,
+                {
+                    scrt2URL: scrt2Url
+                }
+            )
+        }
+    } catch (err) {
+        console.error('Error initializing Embedded Messaging: ', err)
+    }
+}
+
+function useMiaw(
+    scriptLoadStatus,
+    salesforceOrgId,
+    embeddedServiceDeploymentName,
+    embeddedServiceDeploymentUrl,
+    scrt2Url,
+    locale
+) {
+    useEffect(() => {
+        if (scriptLoadStatus.loaded && !scriptLoadStatus.error) {
+            initEmbeddedMessaging(
+                salesforceOrgId,
+                embeddedServiceDeploymentName,
+                embeddedServiceDeploymentUrl,
+                scrt2Url,
+                locale
+            )
+        }
+    }, [
+        scriptLoadStatus,
+        salesforceOrgId,
+        embeddedServiceDeploymentName,
+        embeddedServiceDeploymentUrl,
+        scrt2Url,
+        locale
+    ])
+}
+
+function validateCommerceAgentSettings(commerceAgent) {
+    const requiredFields = [
+        'enabled',
+        'askAgentOnSearch',
+        'embeddedServiceName',
+        'embeddedServiceEndpoint',
+        'scriptSourceUrl',
+        'scrt2Url',
+        'salesforceOrgId',
+        'commerceOrgId',
+        'siteId'
+    ]
+
+    const isValid = requiredFields.every((key) => typeof commerceAgent[key] === 'string')
+    if (!isValid) {
+        console.error('Invalid commerce agent settings.')
+    }
+    return isValid
+}
+
+function isEnabled(enabled) {
+    return enabled === 'true' && onClient
+}
+
+function ShopperAgentWindow({commerceAgentConfiguration, locale, basketId}) {
```

Comment:
Lets try to keep function style consistent with the rest of the application. This means using arrow functions and only using traditional function declarations when required (e.g. you need reference to "this").